pkg/minikube/mount/cifs_windows.go

```diff
@@ -0,0 +1,159 @@
+// +build windows
+
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mount
+
+import (
+	"fmt"
+	"github.com/danieljoos/wincred"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh/terminal"
+	"k8s.io/minikube/pkg/minikube/drivers/hyperv"
+	"k8s.io/minikube/pkg/minikube/out"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+var credentialName = "minikube"
+
+func (m *Cifs) Share() error {
+	// Ensure that the current user is administrator because creating a SMB Share requires Administrator privileges.
+	err := hyperv.IsWindowsAdministrator()
+	if err != nil {
+		return errors.Wrap(err,"current user not administrator")
+	}
+
+	// Check if Name of the Share already exists or not.
+	err = hyperv.PowershellCmd("SmbShare\\Get-SmbShare","-Name",m.HostShareName)
+	if err == nil {
+		glog.Infof("The share with share name %v already exists. Trying to delete it.", m.HostShareName)
+		if err := hyperv.PowershellCmd("SmbShare\\Remove-SmbShare", "-Name", m.HostShareName, "-Force"); err != nil {
+			return errors.Wrap(err, "remove smb share")
+		}
+		glog.Infof("The share with share name %v has been deleted", m.HostShareName)
+	}
+
+	// Get the current user so that we can assign full access permissions to only that user.
+	user, err := hyperv.CurrentWindowsUser()
+	if err != nil {
+		return errors.Wrap(err, "get current user")
+	}
+	glog.Infof("Current User -- [%v]",user)
+	glog.Infof("Trying to enable share for CIFS Mounting.")
+	if err := hyperv.PowershellCmd("SmbShare\\New-SmbShare", "-Name", m.HostShareName, "-Path", m.HostPath , "-FullAccess", user, "-Temporary"); err != nil {
+		return errors.Wrap(err, "new smb share")
+	}
+	return nil
+}
+
+func (m *Cifs) Unshare() error {
+
+	// Ensure that the current user is administrator because creating a SMB Share requires Administrator privileges.
+	err := hyperv.IsWindowsAdministrator()
+	if err != nil {
+		return errors.Wrap(err,"current user not administrator")
+	}
+
+	if err := hyperv.PowershellCmd("SmbShare\\Remove-SmbShare", "-Name", m.HostPath, "-Force"); err != nil {
+		return errors.Wrap(err,"remove smb share")
+	}
+	return nil
+}
+
+func (m *Cifs) Mount(r mountRunner) error {
+	hostname, err := os.Hostname()
```

Comment:
Depending on a specific DNS configuration adds room for failure. Pass in the result of `cluster.GetVMHostIP()`, as the 9p mount does.

This should probably reuse `Config.HostPath` instead of ignoring it.